File: Code Quest/CodeQuest2017/src/Prob06.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.util.HashMap;
import java.util.Map;
public class Prob06 {
	private static final String INPUT_FILE_NAME = "Prob06.in.txt";
	public static void main(String[] args) {
		try {
		    // prepare to read the file
	        File inFile = new File(INPUT_FILE_NAME);
	        FileReader fr = new FileReader(inFile);
	        BufferedReader br = new BufferedReader(fr);
	        String inLine = null;
	        String inLine2 = null;
	        
	        // get the number of test cases
	        int T = Integer.parseInt(br.readLine());
	        Map<Character, String> map = new HashMap<>();
	        map.put('A', "Alpha");
	        map.put('B', "Bravo");
	        map.put('C', "Charlie");
	        map.put('D', "Delta");
	        map.put('E', "Echo");
	        map.put('F', "Foxtrot");
	        map.put('G', "Golf");
	        map.put('H', "Hotel");
	        map.put('I', "India");
	        map.put('J', "Juliet");
	        map.put('K', "Kilo");
	        map.put('L', "Lima");
	        map.put('M', "Mike");
	        map.put('N', "November");
	        map.put('O', "Oscar");
	        map.put('P', "Papa");
	        map.put('Q', "Quebec");
	        map.put('R', "Romeo");
	        map.put('S', "Sierra");
	        map.put('T', "Tango");
	        map.put('U', "Uniform");
	        map.put('V', "Victor");
	        map.put('W', "Whiskey");
	        map.put('X', "Xray");
	        map.put('Y', "Yankee");
	        map.put('Z', "Zulu");
	        
	        while (T > 0) {
	        	// read the line of text
                inLine = br.readLine();
                
                int n = Integer.parseInt(inLine);
                while(n > 0) {
                	inLine2 = br.readLine().toUpperCase();
                	String[] a = inLine2.split("\\s");
                	
                	for(int i = 0; i < a.length; i++) {
                		char[] c = a[i].toCharArray();
                		for(int j = 0; j < c.length; j++) {
                			System.out.print(map.get(c[j]));
                			if(j < c.length -1) {
                				System.out.print("-");
                			} else {
                				System.out.print(" ");
                			}
                		}
                	}
                	System.out.println(" ");
                	n--;
                }
                T--;
	        }
	        
	        br.close();
	        fr.close();
		} catch (Exception e) {
            e.printStackTrace();
        }
	}
}

File: Code Quest/CodeQuest2015/Prob06.py
import re
INPUT_FILE_NAME = "prob06.txt"
def iter(li, l, size):
    for b in range(len(li)):
        for i in range(size[1]):
            for j in range(size[0]):
                if l[j][i] == li[0]:
                    if check(j, i, li, l, 0, size):
                        return True
    return False

def check(c, e, li, l, b, size):
    if b == len(li)-1:
        return True
    if c+1 < size[0]:
        if l[c+1][e] == li[b+1]:
            if check(c+1, e, li, l, b+1, size):
                return True
        if e+1 < size[1]:
            if l[c + 1][e + 1] == li[b + 1]:
                if check(c + 1, e + 1, li, l, b + 1, size):
                    return True
        if e-1 >= 0:
            if l[c + 1][e - 1] == li[b + 1]:
                if check(c + 1, e - 1, li, l, b + 1, size):
                    return True
    if c-1 > 0:
        if l[c - 1][e] == li[b + 1]:
            if check(c - 1, e, li, l, b + 1, size):
                return True
        if e+1 < size[1]:
            if l[c - 1][e + 1] == li[b + 1]:
                if check(c - 1, e + 1, li, l, b + 1, size):
                    return True
        if e-1 >= 0:
            if l[c - 1][e - 1] == li[b + 1]:
                if check(c - 1, e - 1, li, l, b + 1, size):
                    return True
    if e+1 < size[1]:
        if l[c][e + 1] == li[b + 1]:
            if check(c, e + 1, li, l, b + 1, size):
                return True
    if e-1 >= 0:
        if l[c][e - 1] == li[b + 1]:
            if check(c, e - 1, li, l, b+1, size):
                return True
    return False
try:
    f = open(INPUT_FILE_NAME, "r")
    t = int(f.readline())
    size = list(map(int, f.readline().strip("\n").split(" ")))
    print(size)
    l = [[]]*size[1]
    while t > 0:
        t-= 1
        for i in range(size[1]-1):
            l[i] = f.readline().strip("\n").split(" ")
            print(l[i])
    t = int(f.readline())
    while t > 0:
        t-=1
        li = list(filter(None, re.split("",f.readline().strip("\n"))))
        if iter(li, l, size):
            for i in range(len(li)):
                print(li[i], end="")
            print()
finally:
    f.close()